src/region.js

```diff
@@ -33,155 +33,108 @@ var Region = MarionetteObject.extend({
+    const shouldForceShow = !!forceShow;
+    const shouldDestroyView = !preventDestroy;
+    const shouldEmptyRegion = isDifferentView && shouldDestroyView;
+    const shouldShowView = isDifferentView || shouldForceShow;
 
     if (isChangingView) {
       this.triggerMethod('before:swapOut', changingView, this, options);
     }
 
-    if (this.currentView && isDifferentView) {
+    if (isChangingView) {
       delete this.currentView._parent;
     }
 
-    if (_shouldDestroyView) {
-      this.empty();
-
     // A `destroy` event is attached to the clean up manually removed views.
     // We need to detach this event when a new view is going to be shown as it
     // is no longer relevant.
-    } else if (isChangingView && _shouldShowView) {
+    if (!shouldDestroyView || shouldEmptyRegion) {
       this.currentView.off('destroy', this.empty, this);
     }
 
-    if (_shouldShowView) {
-
-      // We need to listen for if a view is destroyed
-      // in a way other than through the region.
-      // If this happens we need to remove the reference
-      // to the currentView since once a view has been destroyed
-      // we can not reuse it.
-      view.once('destroy', this.empty, this);
-
-      // make this region the view's parent,
-      // It's important that this parent binding happens before rendering
-      // so that any events the child may trigger during render can also be
-      // triggered on the child's ancestor views
-      view._parent = this;
-      this._renderView(view, options);
-
-      if (isChangingView) {
-        this.triggerMethod('before:swapIn', view, this, options);
-      }
-
-      this.triggerMethod('before:show', view, this, options);
-      triggerMethodOn(view, 'before:show', view, this, options);
+    // After stopping the listener
+    if (shouldEmptyRegion) {
+      this.empty();
+    }
 
-      // An array of views that we're about to display
-      var attachedRegion = isNodeAttached(this.el);
+    if (!shouldShowView) {
+      return this;
+    }
 
-      // The views that we're about to attach to the document
-      // It's important that we prevent _getNestedViews from being executed unnecessarily
-      // as it's a potentially-slow method
-      var displayedViews = [];
+    this.triggerMethod('before:show', view, this, options);
 
-      var attachOptions = _.extend({
-        triggerBeforeAttach: this.triggerBeforeAttach,
-        triggerAttach: this.triggerAttach
-      }, showOptions);
+    // We need to listen for if a view is destroyed in a way other than through the region.
+    // If this happens we need to remove the reference to the currentView since once a view
+    // has been destroyed we can not reuse it.
+    view.once('destroy', this.empty, this);
 
-      if (attachedRegion && attachOptions.triggerBeforeAttach) {
-        displayedViews = this._displayedViews(view);
-        this._triggerAttach(displayedViews, 'before:');
-      }
+    // Make this region the view's parent.
```

Comment:
Hmm, I don't think so... I think the only concern here is that a parent receives the child's render events triggered during a show, but if there is no render there is no need for those events.  I can not think of a case where we are internally tracking a child's render events.